datadog_checks_base/datadog_checks/base/types.py

```diff
@@ -0,0 +1,38 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Dict, List, Literal, NamedTuple, Optional, Sequence, Tuple, TypedDict, Union
+
+InitConfigType = Dict[str, Any]
+AgentConfigType = Dict[str, Any]
+InstanceType = Dict[str, Any]
+
+ProxySettings = TypedDict(
+    'ProxySettings', {'http': Optional[str], 'https': Optional[str], 'no': List[str]}, total=False
+)
+
+ServiceCheckStatus = Literal[0, 1, 2, 3]
+_ServiceCheckType = NamedTuple(
+    '_ServiceCheckType',
+    [('OK', Literal[0]), ('WARNING', Literal[1]), ('CRITICAL', Literal[2]), ('UNKNOWN', Literal[3])],
+)
+ServiceCheck = _ServiceCheckType(0, 1, 2, 3)
```

Comment:
Does enum work for `ServiceCheck` instead ?

```
class ServiceCheck(Enum):
    OK = 0
    WARNING = 1
    CRITICAL = 2
    UNKNOWN = 3
```


Might need to change:


```
class AgentCheck(object):
    OK = ServiceCheck.OK
    WARNING = ServiceCheck.WARNING
    CRITICAL = ServiceCheck.CRITICAL
    UNKNOWN = ServiceCheck.UNKNOWN

```

Not sure why we have those constant on AgentCheck, it's some kind of duplication.

Reply:
Enum would work from a type checking perspective (enums are static and can't be subclassed so type checkers can interpret them well) but it would change e.g. `ServiceCheck.OK` from `0` to `<ServiceCheck.OK: 0>` (an opaque enum instance), which is potentially breaking.

I agree this implementation can be somewhat jarring though so I added a comment and a test (which I don't think we had before?) for the individual values of this enum-like namedtuple — see 8cd2613007cf3f765ca9645be077c1076ff8b8bf.